### PR TITLE
Add "cs" and "python" categories to the `.quiz` command (and more in desc)

### PR DESCRIPTION
**Pull request by Objectivitix**
## Relevant Issues
Closes issues #753 and #747, extends PR #726

## Description
It was planned to incorporate these two categories in the same PR, however I didn't get enough quiz entries for them. But now, thanks to lak, I have enough! So in total, 60 new questions are added. I also changed the starting embed a bit to make it look nicer, made the variation tolerance stricter, and added some missing answers specified in 747 to the math & science questions.

## Did you:
- [x] Join the [**Python Discord Community**](https://discord.gg/python)?
- [x] Read all the comments in this template?
- [x] Ensure there is an issue open, or link relevant discord discussions?
- [x] Read the [contributing guidelines](https://pythondiscord.com/pages/contributing/contributing-guidelines/)?

### CODE REVIEW

**Comment by Objectivitix** [comment]
lint works, everything works, ready for review :D

**Review by ToxicKidz — CHANGES_REQUESTED** [review]
Hey, you should test before you push, because I get this error when running the code:
```py 
Traceback (most recent call last):
   File "/usr/local/lib/python3.9/site-packages/discord/ext/commands/bot.py", line 618, in _load_from_module_spec
     setup(self)
   File "/bot/bot/exts/evergreen/trivia_quiz.py", line 593, in setup
     bot.add_cog(TriviaQuiz(bot))
   File "/bot/bot/exts/evergreen/trivia_quiz.py", line 202, in __init__
     self.questions = self.load_questions()
   File "/bot/bot/exts/evergreen/trivia_quiz.py", line 222, in load_questions
     return json.loads(p.read_text(encoding="utf-8"))
   File "/usr/local/lib/python3.9/json/__init__.py", line 346, in loads
     return _default_decoder.decode(s)
   File "/usr/local/lib/python3.9/json/decoder.py", line 340, in decode
     raise JSONDecodeError("Extra data", s, end)
 json.decoder.JSONDecodeError: Extra data: line 2 column 10 (char 10)
```

**Comment by Objectivitix** [comment]
@ToxicKidz I did test all 60 questions before, the error arose during the commit (copy paste error smh). It has been resolved :D

File: bot/resources/evergreen/trivia_quiz.json
Comment (ToxicKidz) [inline bot/resources/evergreen/trivia_quiz.json:798]: Isn't [`_sitebuiltins`](https://github.com/python/cpython/blob/main/Lib/_sitebuiltins.py) as file, not a folder?

**Review by ToxicKidz — CHANGES_REQUESTED** [review]

File: bot/resources/evergreen/trivia_quiz.json
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json]: got it, resolved

File: bot/resources/evergreen/trivia_quiz.json
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:904]: hm... lak made these questions, i'll go confirm with him

Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:904]: from python docs:
> Changed in version 3.8: CancelledError is now a subclass of BaseException.

File: bot/resources/evergreen/trivia_quiz.json
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:798]: will change, resolved :D

File: bot/resources/evergreen/trivia_quiz.json
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:904]: i will wait for lak's response before further action

Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:904]: from lak:
> It is a BaseException, not an Exception. Though maybe it should be rephrased to clarify meaning
i will rephrase it to be more precise and less vague

Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json]: resolved :D

File: bot/resources/evergreen/trivia_quiz.json
Comment (Kronifer) [inline bot/resources/evergreen/trivia_quiz.json:375]: Maybe capitalise rationals to keep consistency.

File: bot/resources/evergreen/trivia_quiz.json
Comment (Kronifer) [inline bot/resources/evergreen/trivia_quiz.json:612]: Hyper Text should not be separated by a space.

File: bot/resources/evergreen/trivia_quiz.json
Comment (Kronifer) [inline bot/resources/evergreen/trivia_quiz.json:672]: This could be changed to "Give one protocol that handles mail" as there are SMTP, IMAP and POP3 that are used today

File: bot/resources/evergreen/trivia_quiz.json
Comment (Kronifer) [inline bot/resources/evergreen/trivia_quiz.json:765]: Is this really necessary?

File: bot/resources/evergreen/trivia_quiz.json
Comment (Kronifer) [inline bot/resources/evergreen/trivia_quiz.json:843]: Maybe keep consistency here and enclose multiple choice options in parentheses or do the inverse

File: bot/resources/evergreen/trivia_quiz.json
Comment (Kronifer) [inline bot/resources/evergreen/trivia_quiz.json:884]: there was more than one module added to the stdlib in python 3.9 iirc.

File: bot/resources/evergreen/trivia_quiz.json
Comment (Kronifer) [inline bot/resources/evergreen/trivia_quiz.json:844]: Maybe keep consistency by enclosing multiple choice options in parentheses.

File: bot/resources/evergreen/trivia_quiz.json
Comment (Kronifer) [inline bot/resources/evergreen/trivia_quiz.json:722]: Make sure that you add question marks to the end of questions.

File: bot/resources/evergreen/trivia_quiz.json
Comment (Kronifer) [inline bot/resources/evergreen/trivia_quiz.json:672]: Maybe reword this to "What is the most popular email protocol?" as IMAP and POP3 are still in use.

File: bot/resources/evergreen/trivia_quiz.json
Comment (Kronifer) [inline bot/resources/evergreen/trivia_quiz.json:375]: Maybe capitalise this for consistency, check for capitalisation in other areas as ive noticed this before.

**Review by Kronifer — CHANGES_REQUESTED** [review]
Few nitpicks here

**Comment by Kronifer** [comment]
Github seems to have added duplicate change requests, i did my best to remove them.

File: bot/resources/evergreen/trivia_quiz.json
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:375]: sure, will be resolved in the next commit

File: bot/resources/evergreen/trivia_quiz.json
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:716]: got it

File: bot/resources/evergreen/trivia_quiz.json
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:765]: yes.

File: bot/resources/evergreen/trivia_quiz.json
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:843]: there is consistency - none of these non-lettered ones are enclosed in parentheses

File: bot/resources/evergreen/trivia_quiz.json
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:884]: hm, i'll check with lak

File: bot/resources/evergreen/trivia_quiz.json
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:672]: ok

File: bot/resources/evergreen/trivia_quiz.json
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:884]: lak never responded- i'll think up a new question and replace it

Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json]: resolved

**Comment by Objectivitix** [comment]
@Kronifer @ToxicKidz all your changes have been implemented.

File: bot/resources/evergreen/trivia_quiz.json
Comment (vcokltfre) [inline bot/resources/evergreen/trivia_quiz.json:763]: I see that there's a comment marked as resolved here that asks if it's necessary, which was just answered with 'yes'. While I'm personally unsure whether this should be a question, at the very least I think it's badly phrased, and will lead to confusion, a more suitable question I feel would be along the lines of "Is everything in Python an object?", since "Is it all objects?" sounds too vague.

File: bot/resources/evergreen/trivia_quiz.json
Comment (vcokltfre) [inline bot/resources/evergreen/trivia_quiz.json:778]: This question tripped me up when first reading it because it seems a little ambiguous at first, I think "return when given 2..." would make it more understandable at a glance

**Review by vcokltfre — CHANGES_REQUESTED** [review]

File: bot/resources/evergreen/trivia_quiz.json
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:672]: ah right, to be consistent with "MX, mail exchange" as well.

File: bot/resources/evergreen/trivia_quiz.json
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:651]: got it

File: bot/resources/evergreen/trivia_quiz.json
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:706]: yeah, i agree. it will be refactored

File: bot/resources/evergreen/trivia_quiz.json
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:732]: got it

File: bot/resources/evergreen/trivia_quiz.json
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:763]: i agree with the last part, the question should be phrased better.

File: bot/resources/evergreen/trivia_quiz.json
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:778]: right, you gotta add 'em conjunctions

**Comment by Objectivitix** [comment]
@vcokltfre The changes will be implemented later, currently I gotta update my website lol :D

File: bot/resources/evergreen/trivia_quiz.json
Comment (vcokltfre) [reply bot/resources/evergreen/trivia_quiz.json:651]: This isn't resolved

Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:651]: WAT- i thought i deleted the word lol

Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json]: got it, resolved :D

**Review by vcokltfre — APPROVED** [review]
LGTM! :shipit:

File: bot/resources/evergreen/trivia_quiz.json
Comment (ToxicKidz) [inline bot/resources/evergreen/trivia_quiz.json:626]: Should these underscores be escaped?

**Review by ToxicKidz — CHANGES_REQUESTED** [review]

File: bot/resources/evergreen/trivia_quiz.json
Comment (ToxicKidz) [inline bot/resources/evergreen/trivia_quiz.json:898]: (It seems this comment was not added in the review)
This question doesn't make a lot of sense.

File: bot/resources/evergreen/trivia_quiz.json
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:626]: Oh yeah! I'll do that

File: bot/resources/evergreen/trivia_quiz.json
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:898]: Got it, will replace
~~lak please explain~~

**Review by ToxicKidz — APPROVED** [review]
:shipit:

**Comment by Objectivitix** [comment]
petapoggers :D one last core dev appr then we done

**Review by HassanAbouelela — APPROVED** [review]

**Pull request merged**
